nomad/job_endpoint.go

```diff
@@ -729,69 +730,46 @@ func (j *Job) Deregister(args *structs.JobDeregisterRequest, reply *structs.JobD
 	// Populate the reply with job information
 	reply.JobModifyIndex = index
 
-	evals := []*structs.Evaluation{}
-	now := time.Now().UTC().UnixNano()
+	// Make a raft apply to release the CSI volume claims of terminal allocs.
+	volumesToGC.apply()
```

Comment:
what if there was an error?

Reply:
🤦

Reply:
it wasn't rhetorical, either. i'm really curious what to do if there's an error 😄

Reply:
We should report it to the user, at least. In the case of the `node_endpont.go` path, there's no user waiting for the result, but we can log it and then pick up the claims with the periodic GC.